### PR TITLE
feat:  Add flip augmentation that mirrors training scenes across the ego axis

### PR DESCRIPTION
**Pull request by yamsam**
## Summary
  - Added `FlipAugmentation`, a data augmentation that mirrors the whole scene across the ego longitudinal axis (y → -y) with a per-sample probability
  `flip_prob`
  - Sign flips: y / sin / vy / ay / steering angle / yaw rate of `ego_current_state`; y / sin of `ego_agent_past`, `goal_pose`, `neighbor_agents_past`
  (plus vy), and `static_objects`; y of `polygons` / `line_strings`; y / heading of ego and neighbor futures
  - Side swaps: left/right lane boundaries (`LB_*` ↔ `RB_*` with y negated) and left/right line-type one-hots of `lanes` / `route_lanes` (using the index
  constants from `dimensions.py`); the traffic-light one-hot (green/yellow/red/white/none) is side-agnostic and left untouched
  - `turn_indicators`: `ENABLE_LEFT (2)` ↔ `ENABLE_RIGHT (3)` are swapped, so the turn-indicator GT derived from them stays consistent with the mirrored
  scene
  - Padding rows are all-zero and remain all-zero under both negation and left/right swaps, so no validity masks are needed
  - Wired into `train_epoch` **before** `StatePerturbation` (the flip is a rigid transform, so the downstream perturbation and its collision checks stay
  consistent), with new config fields `use_flip_augment` (default: `False`) and `flip_prob` (default: `0.5`) exposed in `TrainConfig` and the
  `train_predictor.py` CLI. Disabled by default, so existing training setups are unaffected.

  ## Test plan
  - Added `diffusion_planner/tests/test_flip_augmentation.py` (6 tests): per-feature sign flips, boundary/line-type side swaps, turn-indicator swap,
  padding-row preservation, identity at `flip_prob=0`, and double-flip identity
  - `PYTEST_DISABLE_PLUGIN_AUTOLOAD=1 uv run pytest diffusion_planner/tests/test_flip_augmentation.py` — 6 passed
  - Verified `train.py` / `train_epoch.py` import cleanly and the new CLI flags parse into `TrainConfig`

## Visual verification

  `util_scripts/visualize_flip_augmentation.py` verifies that `FlipAugmentation` mirrors scenes correctly, both
  visually and numerically.

  ### Usage

  Run from the `diffusion_planner` directory:

  ```bash
  uv run python util_scripts/visualize_flip_augmentation.py <npz> [<npz> ...] --out_dir <dir>
  ```

  Example with the mini dataset:

  ```bash
  uv run python util_scripts/visualize_flip_augmentation.py \
    ../../mini_datasets/j6_2231_fullseq_mini_20260707/data/**/valid/**/*.npz \
    --out_dir /tmp/flip_check
  ```

### What it does

  For each npz sample, the script reproduces the exact tensor state right before `flip_aug` is applied in
  `train_epoch` (`heading_to_cos_sin` applied to `ego_agent_past` / `goal_pose`, futures kept as `(x, y, heading)`,
  no normalization), applies `FlipAugmentation` with `flip_prob=1.0`, and then:

  1. **Renders a side-by-side PNG** (`<name>_flip_check.png`) of the original and flipped scenes using the existing
  `visualize_inputs`, so lane boundaries, route, goal, neighbors, static objects, and the ego future can be
  inspected at a glance.
  2. **Runs numeric invariant checks**:
     - double flip == identity for every input tensor
     - `y` / heading of ego and neighbor futures are negated
     - flipped left lane boundary == mirrored original right boundary (absolute coordinates)
     - all-zero padding rows stay all-zero
     - turn indicators are swapped `ENABLE_LEFT <-> ENABLE_RIGHT`, other values unchanged

  Each sample is reported as `OK` or `NG: <failed checks>`, and the check result is also stamped on the figure
  title. The script exits with a non-zero code if any check fails, so it can be used as a regression gate in CI or
  after future changes.

## ⚠️  Notes
  - Japan drives on the left; mirrored scenes follow right-hand-traffic conventions (keep-right, mirrored turn geometry). A high `flip_prob` may distort
  the traffic-side prior, so starting with a low value (0.1–0.3) is recommended.
  - The GRPO training path (`train_grpo_predictor.py`) is not wired up in this PR; it can be added the same way if needed.

### CODE REVIEW

**Comment by yamsam** [comment]
Attaching the verification results for the effect of the y flip data augmentation.

13-49-19_00000000_00000493_flip_check
13-49-19_00000000_00000418_flip_check
13-49-19_00000000_00000168_flip_check

**Comment by yamsam** [comment]
Confirmed the run for 10 epochs with mini_datasets.
